modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/internal/processor/test/DefaultSegmentsExperienceRequestProcessorTest.java

```diff
@@ -0,0 +1,200 @@
+			Layout.class.getName());
+
+		Layout layout = _addLayout();
+
+		SegmentsExperience segmentsExperience =
+			_segmentsExperienceLocalService.addSegmentsExperience(
+				segmentsEntry.getSegmentsEntryId(), classNameId,
+				layout.getPlid(), RandomTestUtil.randomLocaleStringMap(), 0,
+				true,
+				ServiceContextTestUtil.getServiceContext(_group.getGroupId()));
+
+		long[] segmentsExperienceIds =
+			_segmentsExperienceRequestProcessor.getSegmentsExperienceIds(
+				new MockHttpServletRequest(), new MockHttpServletResponse(),
+				_group.getGroupId(), classNameId, layout.getPlid(),
+				new long[0]);
+
+		Assert.assertEquals(
+			Arrays.toString(segmentsExperienceIds), 1,
+			segmentsExperienceIds.length);
+
+		Assert.assertEquals(
+			segmentsExperience.getSegmentsExperienceId(),
+			segmentsExperienceIds[0]);
+	}
+
+	@Test
+	public void testGetSegmentsExperienceIdsWithoutSegmentsExperienceIds()
+		throws Exception {
+
+		Layout layout = _addLayout();
+
+		long[] segmentsExperienceIds =
+			_segmentsExperienceRequestProcessor.getSegmentsExperienceIds(
+				new MockHttpServletRequest(), new MockHttpServletResponse(),
+				_group.getGroupId(),
+				_classNameLocalService.getClassNameId(Layout.class.getName()),
+				layout.getPlid(), new long[0]);
+
+		Assert.assertEquals(
+			Arrays.toString(segmentsExperienceIds), 0,
+			segmentsExperienceIds.length);
+	}
+
+	@Test
+	public void testGetSegmentsExperienceIdsWithoutSegmentsExperienceIdsAndWithoutSegmentEntryIds()
+		throws Exception {
+
+		Layout layout = _addLayout();
+
+		long[] segmentsExperienceIds =
+			_segmentsExperienceRequestProcessor.getSegmentsExperienceIds(
+				new MockHttpServletRequest(), new MockHttpServletResponse(),
+				_group.getGroupId(),
+				_classNameLocalService.getClassNameId(Layout.class.getName()),
+				layout.getPlid(), new long[0], new long[0]);
+
+		Assert.assertEquals(
+			Arrays.toString(segmentsExperienceIds), 0,
+			segmentsExperienceIds.length);
+	}
+
+	@Test
+	public void testGetSegmentsExperienceIdsWithSegmentEntryIds()
+		throws Exception {
+
+		SegmentsEntry segmentsEntry = SegmentsTestUtil.addSegmentsEntry(
+			_group.getGroupId());
+
+		long classNameId = _classNameLocalService.getClassNameId(
+			Layout.class.getName());
+
+		Layout layout = _addLayout();
+
+		SegmentsExperience segmentsExperience =
+			_segmentsExperienceLocalService.addSegmentsExperience(
+				segmentsEntry.getSegmentsEntryId(), classNameId,
+				layout.getPlid(), RandomTestUtil.randomLocaleStringMap(), 0,
+				true,
+				ServiceContextTestUtil.getServiceContext(_group.getGroupId()));
+
+		long[] segmentsExperienceIds =
+			_segmentsExperienceRequestProcessor.getSegmentsExperienceIds(
+				new MockHttpServletRequest(), new MockHttpServletResponse(),
+				_group.getGroupId(), classNameId, layout.getPlid(), new long[0],
+				new long[] {segmentsEntry.getSegmentsEntryId()});
+
+		Assert.assertEquals(
+			Arrays.toString(segmentsExperienceIds), 1,
+			segmentsExperienceIds.length);
+
```

Comment:
Not needed

Reply:
I prefer the test fails this way, that with a null pointer exception.

Also, if more experiences are returned, the test should fail too.

Reply:
No, I didn't mean that, It's just the line btween is not needed

Reply:
Usually we add it, when we assert different things, like AssetPublisherUtilTest:92

Reply:
I know, I just saying this because I saw it in the previous PR: https://github.com/liferay/liferay-portal/commit/4b9c90b8510b83abd8d70c632d830c4b959468b6

But, It's not important to be forwarded, of course

Reply:
I think this is a different case, I will wait for Brian's feedback about it, because we would need to refactor many classes with this new pattern otherwise.